File: webflux-dialogue/src/main/java/com/study/webflux/rag/application/monitoring/aop/MonitoredPipeline.java
package com.study.webflux.rag.application.monitoring.aop;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface MonitoredPipeline {
	int inputArgIndex() default 0;
}

File: app-miyou/src/main/kotlin/com/miyou/app/application/monitoring/aop/MonitoredPipeline.kt
package com.miyou.app.application.monitoring.aop

import kotlin.annotation.AnnotationRetention.RUNTIME
import kotlin.annotation.AnnotationTarget.FUNCTION

@Target(FUNCTION)
@Retention(RUNTIME)
annotation class MonitoredPipeline(
    val inputArgIndex: Int = 0,
)
